Add Lookup Table API documentation (#113)
Amplitude-Developer-Docs/amplitude-dev-center: docs/analytics/apis/lookup-tables-api.md

---
title: Lookup Table API
description: The Lookup Table API lets you create, get, update, and delete lookup tables to augment your data.
---

!!!beta "This feature is in beta"

>     This requires that you have the `lookup_table` feature enabled in Amplitude. Ask your rep to help you with this.

Reviewer: ```suggestion
    This requires that you have the `lookup_table` feature enabled in Amplitude. Contact your Amplitude account manager if you need help.
```

### Introduction

Lookup Tables is a way to augment properties. Instead of using formulas, you can upload a CSV file that contains property mappings to derive new properties.

To create a Lookup property, you need to create a Lookup Table to reference. You can retrieve and update each of the tables using the API. Lookup Tables are identified by the name and are scoped per project.

> 

Reviewer: Please add an endpoint section, like this one: https://www.docs.developers.amplitude.com/analytics/apis/behavioral-cohorts-api/#endpoints

```
## Endpoints

| Region | Endpoint |
| --- | --- |
| Standard Server | <ENDPOINT HERE> |
| EU Residency Server | <EU ENDPOINT HERE DELETE ROW IF NOT NEEDED> |
```

### Create a Lookup Table

> To create a Lookup Table, you create a Lookup Table object by uploading a CSV mapping from a property to the new properties to create. You'll need to send a request of type multipart/form-data.

Reviewer: @dnchn I need clarity here: to create a lookup table, the need to send a CSV via request of the multipart/form-data type that maps old properties to the new properties that should be created? 
If so, maybe we can roll with something like this: 

```suggestion
Create a Lookup Table object by uploading a CSV that maps existing properties to the new properties to create. Send the request with the type multipart/form-data type.
```

Reviewer: @dnchn this is the only outstanding thing to get this merged in.

Author: Ah, sorry, I missed this. The suggestion is close, but it can only map a single existing property to the new properties.

|<div class="big-column">Name</div>| Description|
|-----|------|
|`table_name` | Required. String. Name of the table.|
|`file` | Required. File. A CSV representation of the mappings.|

#### Example request

=== "cURL"
    ```curl
    curl -L -X PATCH 'https://amplitude.com/api/2/lookup_table/:table_name' \
         -u API_KEY:SECRET_KEY
         -F 'file=@"/path/to/file.csv"' \
    ```
=== "HTTP"
    ```bash
    PATCH /api/2/lookup_table/:table_name HTTP/1.1
    Host: amplitude.com
    Authorization: Basic {{api-key}}:{{secret-key}}
    Content-Type: multipart/form-data; boundary=----WebKitFormBoundary7MA4YWxkTrZu0gW

    ----WebKitFormBoundary7MA4YWxkTrZu0gW
    Content-Disposition: form-data; name="file"; filename="skuProductReference-updated.csv"
    Content-Type: text/csv

    (data)
    ----WebKitFormBoundary7MA4YWxkTrZu0gW
    ```

#### Response

```json
{
     "name": "skuToMetadata",
     "column_headers": [
         "Product Category",
         "Product Name"
     ],
     "created_at": "2021-07-15T21:04:23.000593",
     "created_by": "rest",
     "last_modified_at": "2021-07-16T19:14:11.627477",
     "last_modified_by": "rest"
}
```

### Delete a Lookup Table

> Delete the Lookup Table

Reviewer: ```suggestion
Delete a Lookup Table.
```
